plugins/sentinelone/komand_sentinelone/tasks/monitor_activities_and_events/task.py

```diff
@@ -0,0 +1,337 @@
+        is_initial_run = not state
+        if is_initial_run:
+            self.logger.info("No state detected. Instantiating initial run.")
+        else:
+            self.logger.info("State detected. Instantiating continuation run.")
+        return is_initial_run
+
+    def check_queries(self, params: Dict) -> Tuple[str, str, str]:
+        """
+        Return whether to query a specific endpoint based on the input parameters
+        """
+        queries = {
+            ACTIVITIES_LOGS: params.get(Input.COLLECTACTIVITIES),
+            EVENTS_LOGS: params.get(Input.COLLECTEVENTS),
+            THREATS_LOGS: params.get(Input.COLLECTTHREATS),
+        }
+        return queries
+
+    def get_cursors(self, state: Dict) -> Tuple[str, str, str]:
+        """
+        Return existing cursors for each endpoint
+        """
+        cursors = {
+            ACTIVITIES_LOGS: state.get(ACTIVITIES_PAGE_CURSOR),
+            EVENTS_LOGS: state.get(EVENTS_PAGE_CURSOR),
+            THREATS_LOGS: state.get(THREATS_PAGE_CURSOR),
+        }
+        return cursors
+
+    def count_queries(self, queries: dict) -> int:
+        "Count how many queries are to be made and return the value"
+        total_queries = 0
+        for query in queries.values():
+            if query is not None:
+                total_queries += 1
+        return total_queries
+
+    def log_pagination_cycle(self, cursors) -> None:
+        """
+        Provide logging information if task is in pagination cycle
+        """
+        pagination_info_string = "Pagination tokens received... "
+        for log_type, cursor in cursors.items():
+            if cursor is not None:
+                pagination_info_string += f" {log_type}:{cursor}"
+        self.logger.info("Paginating cycle in progress.")
+        self.logger.info(pagination_info_string)
+
+    def collect_logs_handler(
+        self,
+        queries: Dict,
+        cursors: Dict,
+        state: Dict,
+        lookback_timestamp: str,
+        total_forbidden_responses: int,
+        is_paginating: bool,
+    ) -> Tuple[List[Dict], int]:
+        """
+        Handle whether to query each endpoint based on input and pagination, gather returned data and error responses
+        """
+        all_logs = []
+        if queries.get(ACTIVITIES_LOGS):
+            activities_logs_next_cursor = cursors.get(ACTIVITIES_LOGS)
+            if is_paginating is False or (is_paginating is True and activities_logs_next_cursor):
+                self.logger.info("Getting activities logs...")
+                activities_logs = self.get_generic_logs(
+                    ACTIVITIES_LOGS,
+                    state,
+                    total_forbidden_responses,
+                    lookback_timestamp,
+                    activities_logs_next_cursor,
+                )
+                all_logs.extend(activities_logs)
+            else:
+                self.logger.info("Pagination in progress, skipping activities log collection...")
```

Comment:
I'm a bit weary that we've saw other tasks get caught on a bad cursor/token loop. Just wondering if we should log out if we're querying using a cursor, just before calling `get_generic_logs`?